src/EditorFeatures/CSharpTest/PdbSourceDocument/PdbSourceDocumentTests.cs

```diff
@@ -594,6 +595,48 @@ public class C
             });
         }
 
+        [Theory]
+        [InlineData(Location.Embedded, "utf-16")]
+        [InlineData(Location.Embedded, "utf-16BE")]
+        [InlineData(Location.Embedded, "utf-32")]
+        [InlineData(Location.Embedded, "utf-32BE")]
+        [InlineData(Location.Embedded, "us-ascii")]
+        [InlineData(Location.Embedded, "iso-8859-1")]
+        [InlineData(Location.Embedded, "utf-8")]
+        [InlineData(Location.OnDisk, "utf-16")]
+        [InlineData(Location.OnDisk, "utf-16BE")]
+        [InlineData(Location.OnDisk, "utf-32")]
+        [InlineData(Location.OnDisk, "utf-32BE")]
+        [InlineData(Location.OnDisk, "us-ascii")]
+        [InlineData(Location.OnDisk, "iso-8859-1")]
+        [InlineData(Location.OnDisk, "utf-8")]
+        public async Task UTF32EmbeddedSource(Location pdbLocation, string encodingWebName)
+        {
+            var source = @"
+public class C
+{
+    public event System.EventHandler E { add { } remove { } }
+}";
+
+            var encoding = Encoding.GetEncoding(encodingWebName);
+            // make sure the encoding webname actually round-trips, or this test will fail for other reasons
+            Assert.Equal(encoding, Encoding.GetEncoding(encoding.WebName));
+
+            await RunTestAsync(async path =>
+            {
+                using var ms = new MemoryStream(encoding.GetBytes(source));
+                var encodedSourceText = EncodedStringText.Create(ms, encoding, canBeEmbedded: true);
+
+                var (project, symbol) = await CompileAndFindSymbolAsync(path, pdbLocation, Location.Embedded, encodedSourceText, c => c.GetMember("C.E"));
+
+                var (actualText, _) = await GetGeneratedSourceTextAsync(project, symbol, expectNullResult: false);
+
+                Assert.NotNull(actualText);
```

Comment:
```suggestion
                AssertEx.NotNull(actualText);
```

This prevents the need for `actualText!.` on the next line

Reply:
Nice! I was disappointed that xunit isn't sufficiently annotated